apps/bfd-pipeline/bfd-pipeline-rif-load/src/main/java/gov/cms/bfd/pipeline/rif/load/RifLoaderIdleTasks.java

```diff
@@ -0,0 +1,355 @@
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import gov.cms.bfd.model.rif.Beneficiary;
+import gov.cms.bfd.model.rif.BeneficiaryHistory;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.BiFunction;
+import javax.crypto.SecretKeyFactory;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.EntityTransaction;
+import org.hibernate.ScrollMode;
+import org.hibernate.ScrollableResults;
+import org.hibernate.Session;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class manages the work that is done during the idle time of a RifLoader. This is a helper
+ * class for the RifLoader. It is expected to have a 1-to-1 relationship to a RifLoader.
+ */
+public class RifLoaderIdleTasks {
+  /** Parameters for the post startup tasks. Tuned for throughput. */
+
+  /** Time slice that a task can take before returning/yielding to the main pipeline */
+  private static final int TASK_TIME_LIMIT_MILLIS = 9800;
+
+  private static final Duration TASK_TIME_LIMIT = Duration.ofMillis(TASK_TIME_LIMIT_MILLIS);
+
+  /** The record count of a db update batch */
+  private static final int BATCH_COUNT = 100;
+
+  /** One thread for each query type */
+  private static final int THREAD_COUNT = 2;
+
+  /** JPQL queries */
+  private static final String SELECT_UNHASHED_BENFICIARIES =
+      "select b from Beneficiary b "
+          + "where b.mbiHash is null and b.medicareBeneficiaryId is not null and "
+          + "b.medicareBeneficiaryId is not empty";
+
+  /** JPQL query for a count */
+  private static final String COUNT_UNHASHED_BENFICIARIES =
+      "select count(b) from Beneficiary b "
+          + "where b.mbiHash is null and b.medicareBeneficiaryId is not null and "
+          + "b.medicareBeneficiaryId is not empty";
+
+  /** JPQL query for a list */
+  private static final String SELECT_UNHASHED_HISTORIES =
+      "select b from BeneficiaryHistory b "
+          + "where b.mbiHash is null and b.medicareBeneficiaryId is not null and "
+          + "b.medicareBeneficiaryId is not empty";
+
+  /** JPQL query for a count */
+  private static final String COUNT_UNHASHED_HISTORIES =
+      "select count(b) from BeneficiaryHistory b "
+          + "where b.mbiHash is null and b.medicareBeneficiaryId is not null and "
+          + "b.medicareBeneficiaryId is not empty";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(RifLoaderIdleTasks.class);
+
+  /** Enum to tell what the current task is being executed. */
+  public enum Task {
```

Comment:
As an alternative have `Task` represent an interface that has a function `do` or something similar. The function `getCurrentTask` would then be able to return a class the implements the `Task` interface and `doCurrentTask` can simply call the `do` function of whatever `getCurrentTask` returns.

Reply:
I understand the thinking behind this suggestion. Here's my thinking... ENUMs are good for representing and testing state. Of course, a class type can represent a state, but doing so would defeat some of the advantages of the interface abstraction. In addition, interfaces require new files in Java. Adding an interface file and the associated implementation files will take something that is fairly simple and splits it across many files. I don't think that is good for readability. If the code became more complex, the extra abstraction would pay off, of course.